dpnp/dpnp_algo/dpnp_algo_arraycreation.pyx

```diff
@@ -54,38 +54,36 @@ __all__ += [
 ]
 
 
-cpdef dparray dpnp_copy(dparray x1, order, subok):
-    return call_fptr_1in_1out(DPNP_FN_COPY, x1, x1.shape)
+ctypedef void(*custom_indexing_1in_1out_func_ptr_t)(void * , void * , const int , size_t * , size_t * , const size_t)
 
 
-cpdef dparray dpnp_diag(v, k):
-    cdef dparray result
+cpdef dparray dpnp_copy(dparray x1, order, subok):
+    return call_fptr_1in_1out(DPNP_FN_COPY, x1, x1.shape)
 
-    # computation of initial position
-    init0 = max(0, -k)
-    init1 = max(0, k)
 
+cpdef dparray dpnp_diag(dparray v, int k):
     if v.ndim == 1:
         size = v.shape[0] + abs(k)
 
-        result = dpnp.zeros(shape=(size, size), dtype=v.dtype)
-
-        for i in range(v.shape[0]):
-            result[init0 + i, init1 + i] = v[i]
-    elif v.ndim == 2:
-        # computation of result size
-        size0 = min(v.shape[0], v.shape[0] + k)
-        size1 = min(v.shape[1], v.shape[1] - k)
-        size = min(size0, size1)
+        shape_result = (size, size)
+    else:
+        size = min(v.shape[0], v.shape[0] + k, v.shape[1], v.shape[1] - k)
         if size < 0:
             size = 0
```

Comment:
@samir-nasibli  looks this https://github.com/IntelPython/dpnp/pull/621/files#diff-d7986221bd1f8eb7a9e1c069323072d09223f2929098ef920f4ed0ba596e391aR70

we have case when, for example, k = -10 and v.shape = (2, 3), then 
size = min(2, 2-10, 3, 3+10)
size = min(2, -8, 3, 13)
=> size == -8 

this out our array, then we need empty array, i.e. size = 0.

Reply:
it looks like the chosen variable name (`size`) is not very good.
Logically variable with name `size` should be non-negative value, it might be a little bit confusing.
I propose to use just `n1` or `n_`.